src/charm.py

```diff
@@ -462,37 +478,24 @@ def _check_and_update_dependencies(self) -> bool:
 
             self._start_services()
             runner_manager.flush(flush_busy=False)
-            self._reconcile_runners(runner_manager)
 
         self.unit.status = ActiveStatus()
         return service_updated or runner_bin_updated
 
-    @catch_charm_errors
-    def _on_update_dependencies(self, _event: UpdateDependenciesEvent) -> None:
-        """Handle checking update of dependencies event.
-
-        Args:
-            event: Event of checking update of runner binary and services.
-        """
-        self._check_and_update_dependencies()
-
     @catch_charm_errors
     def _on_reconcile_runners(self, _event: ReconcileRunnersEvent) -> None:
         """Handle the reconciliation of runners.
 
         Args:
             event: Event of reconciling the runner state.
         """
-        self.unit.status = MaintenanceStatus("Reconciling runners")
+        runner_manager = self._get_runner_manager()
 
-        if not RunnerManager.runner_bin_path.is_file():
-            logger.warning("Unable to reconcile due to missing runner binary")
-            return
+        runner_info = runner_manager.get_github_info()
+        if all(not info.busy for info in runner_info):
+            self._update_kernel(now=True)
```

Comment:
I'm guessing `juju-reboot` will only be triggered after the event handling is done? Otherwise, the check and apply updates should probably happen before the reboot

Reply:
I have update the code here.
The `juju-reboot` has two possible option. One is to reboot after the event handling is done, the other is to reboot immediately. If the `now` arg is true, then it is rebooted immediately.